### PR TITLE
"A secret is revealed!" message

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/hu_stuff.c [hunk 1]
```diff
@@ -1299,11 +1313,24 @@ void HU_Ticker(void)
...
   // display message if necessary
 
-  if ((showMessages || message_dontfuckwithme) && plr->message &&
+  if ((showMessages || message_dontfuckwithme) && plr->centermessage)
+    {
+      extern int M_StringWidth(const char *string);
+      w_secret.l[0].x = ORIGWIDTH/2 - M_StringWidth(plr->centermessage)/2;
+
+      HUlib_addMessageToSText(&w_secret, 0, plr->centermessage);
+      plr->centermessage = 0;
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:1329]: NULL please

File: Source/m_menu.c [hunk 2]
```diff
@@ -2801,14 +2801,15 @@ setup_menu_t stat_settings1[] =  // Status Bar and HUD Settings screen
...
   {"HIDE SECRETS"      ,S_YESNO     ,m_null,ST_X,ST_Y+ 7*8, {"hud_nosecrets"}},
-  {"HEALTH LOW/OK"     ,S_NUM       ,m_null,ST_X,ST_Y+ 8*8, {"health_red"}},
-  {"HEALTH OK/GOOD"    ,S_NUM       ,m_null,ST_X,ST_Y+ 9*8, {"health_yellow"}},
-  {"HEALTH GOOD/EXTRA" ,S_NUM       ,m_null,ST_X,ST_Y+10*8, {"health_green"}},
-  {"ARMOR LOW/OK"      ,S_NUM       ,m_null,ST_X,ST_Y+11*8, {"armor_red"}},
-  {"ARMOR OK/GOOD"     ,S_NUM       ,m_null,ST_X,ST_Y+12*8, {"armor_yellow"}},
-  {"ARMOR GOOD/EXTRA"  ,S_NUM       ,m_null,ST_X,ST_Y+13*8, {"armor_green"}},
-  {"AMMO LOW/OK"       ,S_NUM       ,m_null,ST_X,ST_Y+14*8, {"ammo_red"}},
-  {"AMMO OK/GOOD"      ,S_NUM       ,m_null,ST_X,ST_Y+15*8, {"ammo_yellow"}},
+  {"\"A SECRET IS REVEALED\" MESSAGE",S_YESNO,m_null,ST_X,ST_Y+8*8, {"hud_secret_message"}},
```
Comment (fabiangreffrath) [inline Source/m_menu.c:2804]: Please add this as the last item.

File: Source/m_misc.c [hunk 3]
```diff
@@ -1698,6 +1698,13 @@ default_t defaults[] = {
     "1 to disable display of kills/items/secrets on HUD"
   },
 
+  { // "A secret is revealed" message
+    "hud_secret_message",
+    (config_t *) &hud_secret_message, NULL,
+    {1}, {0,1}, number, ss_none, wad_yes,
```
Comment (fabiangreffrath) [inline Source/m_misc.c:1704]: Please default to 0

Comment (fabiangreffrath) [reply Source/m_misc.c:1704]: wad_yes?

File: Source/p_spec.c [hunk 4]
```diff
@@ -2064,6 +2067,19 @@ void P_PlayerInSpecialSector (player_t *player)
...
+          {
+            int sfx_id;
+
+            player->centermessage = HUSTR_SECRETFOUND;
+
+            sfx_id = I_GetSfxLumpNum(&S_sfx[sfx_secret]) != -1 ? sfx_secret :
+               I_GetSfxLumpNum(&S_sfx[sfx_itmbk]) != -1 ? sfx_itmbk : -1;
+
+            if (sfx_id != -1)
+                S_StartSound(NULL, sfx_itmbk);
```
Comment (fabiangreffrath) [inline Source/p_spec.c:2081]: sfx_id, right?

File: Source/p_spec.c [hunk 5]
```diff
@@ -2064,6 +2067,19 @@ void P_PlayerInSpecialSector (player_t *player)
           // Tally player in secret sector, clear secret special
           player->secretcount++;
           sector->special = 0;
+
+          if (showMessages && hud_secret_message && player == &players[consoleplayer])
+          {
+            int sfx_id;
```
Comment (fabiangreffrath) [inline Source/p_spec.c:2073]: Could we probably make this static and initialize it only once?

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/p_spec.c (hunk 5 above)
Comment (rfomin) [reply Source/p_spec.c:2073]: `static int sfx_id;` ? I don't get it.

**Comment by fabiangreffrath** [comment]
Thank you once again!

File: Source/p_spec.c (hunk 5 above)
Comment (fabiangreffrath) [reply Source/p_spec.c:2073]: `static int sfx_id = ...`. Static variables are only initialized once and then remembered across subsequent function calls. Once we have checked for the sound lumps, we know which one is available and which isn't. No need to check this every time this function is called.

Comment (rfomin) [reply Source/p_spec.c:2073]: In  C static variables only initialized by constant.

Comment (fabiangreffrath) [reply Source/p_spec.c:2073]: Ah, sure, there must have been a reason why I did it like this in Crispy. 😉

**Pull request merged**